lib/appium.js

```diff
@@ -18,6 +18,104 @@ import AsyncLock from 'async-lock';
 import { inspectObject, parseCapsForInnerDriver } from './utils';
 import semver from 'semver';
 
+
+const AUTOMATION_NAMES = {
+  APPIUM: 'Appium',
+  SELENDROID: 'Selendroid',
+  UIAUTOMATOR2: 'UiAutomator2',
+  XCUITEST: 'XCUITest',
+  YOUIENGINE: 'YouiEngine',
+  ESPRESSO: 'Espresso',
+  FAKE: 'Fake',
+};
+const DRIVER_MAP = {
+  SelendroidDriver: {
+    driverClass: SelendroidDriver,
+    moduleName: 'appium-selendroid-driver',
+    automationName: AUTOMATION_NAMES.SELENDROID,
+  },
+  AndroidUiautomator2Driver: {
+    driverClass: AndroidUiautomator2Driver,
+    moduleName: 'appium-uiautomator2-driver',
+    automationName: AUTOMATION_NAMES.UIAUTOMATOR2,
+  },
+  XCUITestDriver: {
+    driverClass: XCUITestDriver,
+    moduleName: 'appium-xcuitest-driver',
+    automationName: AUTOMATION_NAMES.XCUITEST,
+  },
+  YouiEngineDriver: {
+    driverClass: YouiEngineDriver,
+    moduleName: 'appium-youiengine-driver',
+    automationName: AUTOMATION_NAMES.YOUIENGINE,
+  },
+  FakeDriver: {
+    driverClass: FakeDriver,
+    moduleName: 'appium-fake-driver',
+  },
+  AndroidDriver: {
+    driverClass: AndroidDriver,
+    moduleName: 'appium-android-driver',
+  },
+  IosDriver: {
+    driverClass: IosDriver,
+    moduleName: 'appium-ios-driver',
+  },
+  WindowsDriver: {
+    driverClass: WindowsDriver,
+    moduleName: 'appium-windows-driver',
+  },
+  MacDriver: {
+    driverClass: MacDriver,
+    moduleName: 'appium-mac-driver',
+  },
+  EspressoDriver: {
+    driverClass: EspressoDriver,
+    moduleName: 'appium-espresso-driver',
+    automationName: AUTOMATION_NAMES.ESPRESSO,
+  },
+  TizenDriver: {
+    driverClass: TizenDriver,
+    moduleName: 'appium-tizen-driver',
+    automationName: AUTOMATION_NAMES.TIZEN,
+  },
+};
+// load the version for each driver
+for (const driverProps of _.values(DRIVER_MAP)) {
```

Comment:
doesn't feel super `const`y to be setting properties on this, and mutating what's inside `DRIVER_MAP` /shrug

Reply:
Adjusted for Moar Constantiness.